### PR TITLE
doc: de-duplicate instructions in kind and hubble getting started guides

### PR DESCRIPTION
**Pull request by rolinh**
These 2 guides have a lot of instructions in common as the Hubble guide
instructs to create a sandbox environment using Kind. This commit
ensures that instructions are not duplicated in the two guides but are
included from separate files from the 2 guides.

In addition to this, this commit also updates the instructions about
possible conflict of IP address when deploying a cluster with Kind: the
"tip" has been moved to the "Configure Kind" section as the fix needs to
be applied when configuring Kind.

Moreover, remove the "Optional" wording with regard to pre-loading the
cilium image into the Kind cluster.

### CODE REVIEW

File: Documentation/gettingstarted/kind.rst [hunk 1]
```diff
@@ -48,19 +39,8 @@ Install Cilium release via Helm:
...
-   and re-create the cluster.
-
-   .. parsed-literal::
-        networking:
-          disableDefaultCNI: true
-          podSubnet: "10.244.0.0/16"
-          serviceSubnet: "10.245.0.0/16"
-
 .. include:: k8s-install-validate.rst
+
```
Comment (michi-covalent) [inline Documentation/gettingstarted/kind.rst]: nit: remove this blank line?

File: Documentation/gettingstarted/kind-install.rst [hunk 2]
```diff
@@ -13,36 +13,82 @@ Install Dependencies
...
 .. code:: bash
 
     kind create cluster --config=kind-config.yaml
 
-This will add a ``kind-kind`` context to ``KUBECONFIG`` or if unset,
-``${HOME}/.kube/config``
+After a couple of seconds or minutes, a 4 nodes cluster should be created.
+
+.. parsed-literal::
+    $ kind create cluster --config kind-config.yaml
```
Comment (michi-covalent) [inline Documentation/gettingstarted/kind-install.rst]: nit: i have a mixed feeling about including command output in the documentation. it does take up quite a bit of space :)

**Review by michi-covalent — APPROVED** [review]
lgtm, a couple of non-blocking minor nits

File: Documentation/gettingstarted/kind-install.rst (hunk 2 above)
Comment (joestringer) [reply Documentation/gettingstarted/kind-install.rst]: nit: If we do, we should use `.. code-block:: shell-session` to make the syntax highlighting work correctly (unless there's actually parsed literals like the chart version variable or similar.)

File: Documentation/gettingstarted/kind-install.rst [hunk 3]
```diff
@@ -13,36 +13,82 @@ Install Dependencies
...
 
-Start Kind
-==========
+    If any of these subnets conflicts with your local network address range,
+    update the ``networking`` section of the kind configuration file to specify
+    different subnets that do not conflict or you risk having connectivity
+    issues when deploying Cilium. For example:
 
-Pass the ``kind-config.yaml`` you created with the ``--config`` flag of kind.
+    .. parsed-literal::
```
Comment (joestringer) [inline Documentation/gettingstarted/kind-install.rst]: nit: This can be `::` or `.. code-block:: yaml` (I think)

File: Documentation/gettingstarted/kind-preload.rst [hunk 4]
```diff
@@ -0,0 +1,8 @@
+As the ``cilium`` pod is deployed on every node, let's preload the ``cilium``
+image into the kind cluster so each worker doesn't have to pull it
+individually:
+
+.. parsed-literal::
+
+  docker pull cilium/cilium:|IMAGE_TAG|
+  kind load docker-image cilium/cilium:|IMAGE_TAG|
```
Comment (joestringer) [inline Documentation/gettingstarted/kind-preload.rst:6]: This is the kind of situation where `parsed-literal` is required.

File: Documentation/gettingstarted/hubble.rst [hunk 5]
```diff
@@ -192,17 +170,17 @@ port-forward the Hubble Relay service locally:
    This terminal window needs to be remain open to keep port-forwarding in
    place. Open a separate terminal window to use the ``hubble`` CLI.
 
-You may test that your setup is working by issuing this command which gives you
-the status of the Hubble Relay service:
+You may verify that your setup is working by issuing this command which gives
+you the status of the Hubble Relay service:
```
Comment (joestringer) [inline Documentation/gettingstarted/hubble.rst]: nit:

```suggestion
Confirm that the Hubble Relay service is healthy via `hubble status`:
```

File: Documentation/gettingstarted/kind.rst [hunk 6]
```diff
@@ -23,17 +22,9 @@ Install Cilium
 ==============
 
 .. include:: k8s-install-download-release.rst
+.. include:: kind-preload.rst
```
Comment (joestringer) [inline Documentation/gettingstarted/kind.rst:25]: Is this file intended to either include or point to the `kind-install.rst`? I missed where that takes place.

File: Documentation/gettingstarted/kind-preload.rst [hunk 7]
```diff
@@ -0,0 +1,8 @@
+As the ``cilium`` pod is deployed on every node, let's preload the ``cilium``
+image into the kind cluster so each worker doesn't have to pull it
+individually:
```
Comment (joestringer) [inline Documentation/gettingstarted/kind-preload.rst]: nit: 
```suggestion
Preload the ``cilium`` image into each worker node in the kind cluster:
```

**Review by joestringer — CHANGES_REQUESTED** [review]
Looks good, thanks for tidying this up.

A made some minor suggestions below to use imperative sense to simplify the language on the steps necessary to install Helm. (I realise they're not all from this PR, but I think that they clarify the wording).

I also made some suggestions on the use of sphinx directives for syntax highlighting, things like `.. parsed-literal` usage when literals are not passed.

File: Documentation/gettingstarted/kind-install.rst (hunk 2 above)
Comment (rolinh) [reply Documentation/gettingstarted/kind-install.rst]: OK let's remove this.

File: Documentation/gettingstarted/kind.rst (hunk 6 above)
Comment (rolinh) [reply Documentation/gettingstarted/kind.rst:25]: This is just the instructions to pre-load the cilium image into the kind cluster. As this is required for both the Hubble and Kind getting started guide, I've split this into a separate file that is included in both gsgs.

File: Documentation/gettingstarted/kind-install.rst (hunk 3 above)
Comment (rolinh) [reply Documentation/gettingstarted/kind-install.rst]: Thanks for the tip, `.. code-block:: yaml` is the more appropriate in this case.

File: Documentation/gettingstarted/kind.rst (hunk 6 above)
Comment (joestringer) [reply Documentation/gettingstarted/kind.rst:25]: Not sure how I missed it before, now I look it's definitely there. :+1:

**Review by joestringer — APPROVED** [review]
LGTM, looks like the CI is just complaining on one minor spelling point because it doesn't know the word `Preload`.
